lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -3723,6 +3723,132 @@ class ConvertAtenBroadcastToOp : public OpConversionPattern<AtenBroadcastToOp> {
 };
 } // namespace
 
+// Collapses a `tensor` of rank k into a `tensor` of rank 1.
+// Special case 0d tensor, in that case it's an expansion.
+static Value collapseTo1dTensor(OpBuilder &b, Location loc, Value tensor) {
+  auto tensorType = tensor.getType().cast<RankedTensorType>();
+  if (tensorType.getRank() == 0) {
+    Value one = b.create<arith::ConstantIndexOp>(loc, 1);
+    Value scalarVal = b.create<tensor::ExtractOp>(loc, tensor);
+    return createInitTensor(b, loc, {one}, tensorType.getElementType(),
+                            scalarVal);
+  }
+  if (tensorType.getRank() == 1)
+    return tensor;
+  SmallVector<ReassociationIndices> reassociation(1);
+  for (auto i : llvm::seq<int64_t>(0, tensorType.getRank()))
+    reassociation[0].push_back(i);
+  return b.create<linalg::TensorCollapseShapeOp>(loc, tensor, reassociation);
```

Comment:
Please wait for Garvey's PR to merge. You will have to change it to `<TensorCollapeShapeOp>` instead.

Reply:
I see, thanks.